packages/calypso-products/src/plans.tsx

```diff
@@ -6,30 +6,53 @@ import { getPlansListExperiment } from './experiments';
  * See: p7H4VZ-4S4-p2
  */
 
+const isPlanNameChangeTreatment =
+	getPlansListExperiment( 'wpcom_plan_name_change_personal_premium_v1' ) === 'treatment';
```

Comment:
This actually won't work as the `getPlansListExperiment` gets called before `setPlansListExperiment` has the change to set the correct experiment from the API.
Keeping the `getPlansListExperiment` inside the `get{PLAN_NAME}Title` should delay the assignment lookup until the last moment.

Reply:
True that. I've fixed it here https://github.com/Automattic/wp-calypso/pull/91381